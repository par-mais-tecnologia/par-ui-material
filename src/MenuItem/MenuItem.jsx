import React, { PureComponent } from 'react'
import MenuItemMUI from '@material-ui/core/MenuItem'

class MenuItem extends PureComponent {
  render () {
    return (
      <MenuItemMUI {...this.props} />
    )
  }
}

export default MenuItem
