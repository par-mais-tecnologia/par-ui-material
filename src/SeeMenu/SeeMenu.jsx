import React from 'react'

import Button from '../Button'
import Typography from '@material-ui/core/Typography'

import {
  Menu,
  MenuItem,
  withStyles
} from '@material-ui/core'

const styles = theme => (theme.menu)

const SeeMenu = ({
  name,
  lastUpdate,
  menuItems,
  handleClose,
  handleExit,
  isOpen,
  className,
  selectedIndex,
  classes
}) => (
  <Menu
    id='simple-menu'
    open={isOpen}
    onClose={handleClose}
    disableAutoFocusItem
    anchorReference='anchorPosition'
    anchorPosition='left'
    className={className}
  >
    <div className='pl3 pt2'>
      <i className='gray par-icon-close' onClick={handleClose} />
    </div>
    <div className='flex flex-column mv4 ph3 items-center'>
      <Typography className='pb3' variant='overline'>
        {`${name}`}
      </Typography>
      <Button
        variant='contained'
        onClick={handleExit}>
        SAIR
      </Button>
    </div>
    <div className='mt1 mh3 bt bb bw-1' style={{ borderColor: '#E2E2E2' }}>
      {
        [...menuItems].map((item, index) => (
          <MenuItem
            onClick={item.onClick}
            disableGutters
            classes={{ root: classes.menuItem }}
            key={`menuItemMobile-${item.item}`}
          >
            <Typography
              variant='caption'
              className='w-100'
              color='inherit'
              style={selectedIndex === index
                ? { color: '#347A7C', fontFamily: '\'Roboto Bold\', sans-serif' }
                : {}}
            >
              {item.item}
            </Typography>
          </MenuItem>
        ))
      }
    </div>
    <div className='flex flex-column ph3 mv3'>
      <Typography variant='overline'>AMBIENTE ATUALIZADO DIARIAMENTE</Typography>
      <Typography variant='body1'>Última atualização</Typography>
      <Typography variant='body1'>{`${lastUpdate.date} às ${lastUpdate.time}`}</Typography>
    </div>
  </Menu>
)

export default withStyles(styles)(SeeMenu)
