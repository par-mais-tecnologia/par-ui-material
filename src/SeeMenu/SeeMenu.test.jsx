import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import SeeMenu from './'
import MenuItem from '@material-ui/core/MenuItem'

configure({ adapter: new Adapter() })

const menuItems = [
  {
    item: 'SEUS OBJETIVOS',
    onClick: () => {}
  },
  {
    item: 'INTELIGÊNCIA EM INVESTIMENTOS',
    onClick: () => {}
  },
  {
    item: 'INFORMAÇÕES',
    onClick: () => {}
  },
  {
    item: 'CONTATO',
    onClick: () => {}
  }
]

export const testShallow = describe('SeeMenu shallow', () => {
  const getShallow = () => (
    <SeeMenu
      name='Rui Cavaleiro'
      lastUpdate={{ date: '29/09/1989', time: '14:34' }}
      menuItems={menuItems}
      isOpen
      handleClose={() => {}}
    />
  )

  it(`shallow Stepper`, () => {
    const wrapper = shallow(getShallow())
    expect(wrapper).toBeTruthy()
  })

  it('shallow Stepper match Snapshot', () => {
    const wrapper = shallow(getShallow())
    expect(wrapper).toMatchSnapshot()
  })

  it(`should have the number of <MenuItem/>'s equals the number of steps`, () => {
    const wrapper = mount(<SeeMenu
      name='Rui Cavaleiro'
      lastUpdate={{ date: '29/09/1989', time: '14:34' }}
      menuItems={menuItems}
      isOpen
      handleClose={() => {}}
    />)
    expect(wrapper.find(MenuItem)).toHaveLength(menuItems.length)
  })
})
