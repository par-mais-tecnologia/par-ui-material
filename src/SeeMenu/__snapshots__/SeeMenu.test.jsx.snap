// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeeMenu shallow shallow SeeMenu match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SeeMenu)
    handleClose={[Function]}
    isOpen={true}
    menuItems={
      Array [
        Object {
          "item": "SEUS OBJETIVOS",
          "onClick": [Function],
        },
        Object {
          "item": "INTELIGÊNCIA EM INVESTIMENTOS",
          "onClick": [Function],
        },
        Object {
          "item": "INFORMAÇÕES",
          "onClick": [Function],
        },
        Object {
          "item": "CONTATO",
          "onClick": [Function],
        },
      ]
    }
    mobileFooterBody="Última atualização 
20/09/1989"
    mobileFooterTitle="Ambiente atualizado diariamente"
    name="Rui Cavaleiro"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {},
      "handleClose": [Function],
      "isOpen": true,
      "menuItems": Array [
        Object {
          "item": "SEUS OBJETIVOS",
          "onClick": [Function],
        },
        Object {
          "item": "INTELIGÊNCIA EM INVESTIMENTOS",
          "onClick": [Function],
        },
        Object {
          "item": "INFORMAÇÕES",
          "onClick": [Function],
        },
        Object {
          "item": "CONTATO",
          "onClick": [Function],
        },
      ],
      "mobileFooterBody": "Última atualização 
20/09/1989",
      "mobileFooterTitle": "Ambiente atualizado diariamente",
      "name": "Rui Cavaleiro",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {},
        "handleClose": [Function],
        "isOpen": true,
        "menuItems": Array [
          Object {
            "item": "SEUS OBJETIVOS",
            "onClick": [Function],
          },
          Object {
            "item": "INTELIGÊNCIA EM INVESTIMENTOS",
            "onClick": [Function],
          },
          Object {
            "item": "INFORMAÇÕES",
            "onClick": [Function],
          },
          Object {
            "item": "CONTATO",
            "onClick": [Function],
          },
        ],
        "mobileFooterBody": "Última atualização 
20/09/1989",
        "mobileFooterTitle": "Ambiente atualizado diariamente",
        "name": "Rui Cavaleiro",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
